packages/brisa/src/utils/rpc/rpc.ts

```diff
@@ -113,6 +113,12 @@ function spaNavigation(event: any) {
   ) {
     event.intercept({
       async handler() {
+        // Prevent navigation if the destination URL is the same as the current location
+        if (event.destination.url === location.href) {
```

Comment:
Currently it does not enter here because [`event.canIntercept`](https://developer.mozilla.org/en-US/docs/Web/API/NavigateEvent/canIntercept) is `false` in this case, as MDN says:

> For other URL types, navigation can be intercepted if only the fragment part differs.

fragment part doesn't differs, so `canIntercept` is `false`.

To fix this probably we can create a variable outside with this `event.destination.url === location.href` condition, and check together with the `canIntercept` on the above condition, and then here to skip the fetch + diffing.

Reply:
Have I handled the logic correctly @aralroca ?